.github/workflows/tests.yml

```
@@ -1,51 +1,36 @@
 name: continuous-integration
 
-on: [push, pull_request]
+on:
+  push:
+    branches: [master]
+  pull_request:
 
 jobs:
   pre-commit:
-
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.8
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.8
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -e .[code_style]
-    - name: Run pre-commit
-      run: |
-        pre-commit run --all-files || ( git status --short ; git diff ; exit 1 )
-
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
```

is there a reason to not be using v2 here, and also specifying the python version

---

Doing what's recommended in https://github.com/pre-commit/action#using-this-action. I remember Anthony mentioning that there's something about v2 that makes it tricky to keep things concise.

Not mentioning the python version results in the step getting the latest Python 3 version (basically same as saying "3.x").

---

```suggestion
    - uses: actions/setup-python@v1
      with:
        python-version: 3.8
```

It's a nit-pick, but just reduces the possibility of the tests suddenly failing because a new python version is released

---

They won't, but 🤷🏽

---

Well the reason I say this is because on my other project all the tests started failing the other day due to a change in python version in `setup-python`: https://github.com/actions/virtual-environments/issues/1202. So never say never 😉